javascript/ql/lib/semmle/javascript/DOM.qll

```diff
@@ -420,6 +420,13 @@ module DOM {
     t.startInProp("target") and
     result = domEventSource()
     or
+    t.start() and
+    exists(DataFlow::ClassNode cls |
+      cls.getASuperClassNode().getALocalSource() =
+        DataFlow::globalVarRef(any(string s | s.matches("HTML%Element"))) and
+      result = cls.getAReceiverNode()
+    )
+    or
```

Comment:
How about using `ClassNode::getAnInstanceReference()`, and placing it in `domValueRef()` below?